devicetypes/smartthings/samsung-smart-doorlock.src/samsung-smart-doorlock.groovy

```diff
@@ -0,0 +1,402 @@
+/**
+ *  Samsung Smart Doorlock
+ *
+ *  Copyright 2018 Samsung SDS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License. You may obtain a copy of the License at:
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed
+ *  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License
+ *  for the specific language governing permissions and limitations under the License.
+ *
+ */
+import physicalgraph.zigbee.zcl.DataType
+
+
+metadata {
+	definition (name: "Samsung Smart Doorlock", namespace: "Samsung SDS", author: "kyun.park", mnmn: "SmartThings", vid: "generic-lock") {
+		capability "Actuator"
+		capability "Lock"
+		capability "Battery"
+		capability "Configuration"
+        capability "Health Check"
+        capability "Refresh"
+
+        fingerprint profileId: "0104", inClusters: "0000, 0001, 0003, 0004, 0005, 0009, 0101", outClusters: "0019", manufacturer: "SAMSUNG SDS"
```

Comment:
Does the device populate the model name attribute? If it does the model should be listed here in the fingerprint.

Reply:
At the time of zigbee join, the model name comes in as an empty string. Therefore, it was not included as a part of the fingerprint.